plugins/module_utils/network/sonic/config/snmp/snmp.py

```diff
@@ -0,0 +1,1545 @@
+        elif state == 'merged':
+            commands, requests = self._state_merged(want, have)
+
+        return commands, requests
+
+    def _state_replaced_or_overridden(self, want, have, state):
+        """
+        The command generator when state is overridden
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+
+        if not want:
+            return commands, requests
+
+        diff_want = get_diff(want, have, TEST_KEYS)
+        del_commands = get_diff(have, want, TEST_KEYS)
+        if not diff_want and not del_commands:
+            return commands, requests
+        merged_commands = None
+        merged_request = None
+
+        if del_commands:
+            if state == 'replaced':
+                pop_list = []
+                for option in del_commands:
+                    if not want.get(option):
+                        pop_list.append(option)
+                for option in pop_list:
+                    del_commands.pop(option)
+            if 'agentaddress' in del_commands and 'agentaddress' in want:
+                for command in del_commands.get('agentaddress'):
+                    # If no differences were found other than 'name' (which is auto-generated for the config),
+                    # don't delete this entry.
+                    if len(command) == 1 and command.get('name'):
+                        del_commands['agentaddress'].remove(command)
+                if len(del_commands['agentaddress']) == 0:
+                    del_commands.pop('agentaddress')
+            if 'host' in del_commands and 'host' in want:
+                delete_these_hosts = []
+                for command in del_commands.get('host'):
+                    # If no differences were found other than 'name' (which is auto-generated for the config),
+                    # don't delete this entry.
+                    new_command = {}
+                    if (((len(command) == 1 and 'name' in command) or (len(command) == 2 and 'name' in command and 'ip' in command)
+                         or (len(command) == 3 and 'ip' in command and 'port' in command and 'name' in command))):
+                        break
+                    else:
+                        new_command = command
+
+                    if (new_command):
+                        delete_these_hosts.append(new_command)
+
+                if len(delete_these_hosts) == 0:
+                    del_commands.pop('host')
+                else:
+                    del_commands['host'] = delete_these_hosts
+            if len(del_commands) > 0:
+                del_requests = self.get_delete_snmp_request(del_commands, have, False)
+                if del_requests:
+                    requests.extend(del_requests)
+                    commands.extend(update_states(del_commands, "deleted"))
+
```

Comment:
Depending on how you decide to handle host deletion for hosts with 'diffs' in replaced/overridden states, diff_want may need to be adjusted here by subtract hosts in "delete_these_hosts" from "have" and re-calculating "diff_want" before doing the next two instructions to account for any hosts that were deleted, but have configuration in 'want'. (e.g. Remove those hosts from 'have' and recalculate the diff.)